api/src/test/java/org/openmrs/validator/PersonValidatorTest.java

```diff
@@ -197,5 +198,35 @@ public void shouldNotSetDeathBeforeBirth() {
 		Assert.assertTrue(errors.hasFieldErrors("deathDate"));
 	}
 
+	@Test
+	public void validate_shouldFailValidationWhenDeathCauseAndDeathCauseNonCodedAreSet() {
+		Patient pa = new Patient(1);
+		pa.setDead(true);
+		
+		pa.setCauseOfDeathNonCoded("Some text describing Cause of Death");
+		pa.setCauseOfDeath(new Concept());
+		
+		Errors errors = new BindException(pa, "patient");
+		validator.validate(pa, errors);
+		
+		Assert.assertTrue(errors.hasFieldErrors("causeOfDeath"));
+	}
+	
+	@Test
+	public void validate_shouldNotFailWhenDeathCauseNotCodedIsSet() {
+		Person person = new Patient(1);
+		
+		person.setDead(true);
+		person.setCauseOfDeathNonCoded("Some text describing Cause of Death");
+		person.setPersonVoided(true);
+		person.setPersonVoidReason("voidReason");
+		person.setGender("g");
+		
+		Errors errors = new BindException(person, "patient");
+		PersonValidator personValidator = new PersonValidator();
+		personValidator.validate(person, errors);
+
+		Assert.assertFalse(errors.hasErrors());
+	}
```

Comment:
Where is the test that ensures the test fails if both fields are not set

Reply:
I think this case was already implemented by:
validate_shouldFailValidationIfCauseOfDeathIsBlankWhenPatientIsDead

That's why I did not add it

Reply:
But that test is making the check for one field and not both

Reply:
Depends what test you are refering to. The test _validate_shouldNotFailWhenDeathCauseNotCodedIsSet_ which you commented to is  making the check for one field.

However, there already existed a test case before which neither set _causeOfDeath_ nor **causeOfDeathNonCoded**. As far as I can see and understand this code, this test case is called: **validate_shouldFailValidationIfCauseOfDeathIsBlankWhenPatientIsDead** and should validate the case if both are not set.
---> https://github.com/openmrs/openmrs-core/blob/1327f91d828cfa81f9412f6cf29137d21de6a519/api/src/test/java/org/openmrs/validator/PersonValidatorTest.java#L216


Or how would you expect that test to check or behave if both fields are not set

Reply:
I think that test satisfies it then